src/tools/kdb/import.cpp

```diff
@@ -87,44 +106,41 @@ int ImportCommand::execute (Cmdline const & cl)
 		return 0;
 	}
 
-	KeySet base = originalKeys.cut (root);
+	kdb::KeySet base = originalKeys.cut (root);
 	importedKeys = importedKeys.cut (root);
 	if (cl.withoutElektra)
 	{
-		KeySet baseCopy = base.dup ();
-		Key systemElektra ("system/elektra", KEY_END);
-		KeySet systemKeySet = baseCopy.cut (systemElektra);
+		kdb::KeySet baseCopy = base.dup ();
+		kdb::Key systemElektra ("system/elektra", KEY_END);
+		kdb::KeySet systemKeySet = baseCopy.cut (systemElektra);
 		importedKeys.append (systemKeySet);
 	}
-
-	ThreeWayMerge merger;
-	MergeHelper helper;
-
-	helper.configureMerger (cl, merger);
-	MergeResult result = merger.mergeKeySet (
-		MergeTask (BaseMergeKeys (base, root), OurMergeKeys (base, root), TheirMergeKeys (importedKeys, root), root));
-
-	helper.reportResult (cl, result, cout, cerr);
-
-	int ret = -1;
-	if (!result.hasConflicts ())
+	ckdb::Key * informationKey = ckdb::keyNew (0, KEY_END);
+	ckdb::KeySet * resultNew = elektraMerge (base.getKeySet (), root.getKey (), importedKeys.getKeySet (), root.getKey (),
```

Comment:
Yes, the concept looks good. Missing is, however, getting rid of these special cases like `cl.strategy == "validate"`. This should be integrated in a elektraMergeResult or elektraMergeTool API. It can be in a later PR though. For editing tools (like qt-gui) elektraMerge is enough.